pkg/command/command.go

```diff
@@ -159,22 +162,80 @@ func startCmd(cmd *exec.Cmd, _out, _err io.Writer) (*execution, error) {
 	execution := execution{cmd: cmd}
 	execution.wg.Add(2)
 
+	srcOut := stdout
+	srcErr := stderr
+
+	if c.ErrorCategoryMapping != nil {
+		prOut, pwOut := io.Pipe()
+		trOut := io.TeeReader(stdout, pwOut)
+		srcOut = prOut
+
+		prErr, pwErr := io.Pipe()
+		trErr := io.TeeReader(stderr, pwErr)
+		srcErr = prErr
+
+		execution.wg.Add(2)
+
+		go func() {
+
+			defer execution.wg.Done()
+			defer pwOut.Close()
+
+			scanner := bufio.NewScanner(trOut)
+			for scanner.Scan() {
+				line := scanner.Text()
+				c.parseConsoleErrors(line)
+			}
+			if err := scanner.Err(); err != nil {
+				log.Entry().WithError(err).Info("failed to scan log file")
+			}
+
+		}()
+
+		go func() {
+
+			defer execution.wg.Done()
+			defer pwErr.Close()
+
+			scanner := bufio.NewScanner(trErr)
```

Comment:
I have the suspicion that this will again introduce the problem with the maximum of 64KiB output buffers in case where tool output is longer than 64KiB between '\n'.

Reply:
Any idea how we could avoid this one?

Reply:
from the `bufio/Scanner` docs

> Scanning stops unrecoverably at EOF, the first I/O error, or a token too large to fit in the buffer. When a scan stops, the reader may have advanced arbitrarily far past the last token. Programs that need more control over error handling or large tokens, or must run sequential scans on a reader, should use bufio.Reader instead.

From what I understand here the worst happening could be that the parsing is not 100% as accepted which I would accept ...

Reply:
what do you think?

Reply:
Sorry for the late reply... I think you could be correct. Perhaps this can simply be checked/confirmed with a test. Maybe similar the log/writer test.

Also, an alternative is to provide your own tokenizing function. The problem with the default writer from the log was that you had no way of overriding the default tokenizing function, since the scanner was used somewhere deep in the internals. But here, you are using it directly, so you have that option.